src/client/SquareClient.ts

```diff
@@ -69,25 +73,25 @@ export class SquareClient {
     }
 
     getApplePayApi(retryableMethods: string[] = []): ApplePayApi {
-        return this.proxy(new ApplePayApi(this.getOriginClient()), retryableMethods);
+        return this.proxy('applePayApi', retryableMethods);
     }
 
     getCatalogApi(
         retryableMethods: string[] = ['batchRetrieveCatalogObjects', 'catalogInfo', 'listCatalog', 'retrieveCatalogObject', 'searchCatalogObjects'],
     ): CatalogApi {
-        return this.proxy(new CatalogApi(this.getOriginClient()), retryableMethods);
+        return this.proxy('catalogApi', retryableMethods);
     }
 
     getCheckoutApi(retryableMethods: string[] = []): CheckoutApi {
-        return this.proxy(new CheckoutApi(this.getOriginClient()), retryableMethods);
+        return this.proxy('checkoutApi', retryableMethods);
     }
 
     getCustomersApi(retryableMethods: string[] = ['listCustomers', 'retrieveCustomer', 'searchCustomers', 'deleteCustomerCard']): CustomerClientApi {
-        return this.proxy(new CustomerClientApi(this.getOriginClient()), retryableMethods);
+        return this.proxyWithInstance('customersApi', new CustomerClientApi(this.getOriginClient()), retryableMethods);
```

Comment:
Why this one is different?

Reply:
we have our own custom `CustomerClientApi` class with some additional logic